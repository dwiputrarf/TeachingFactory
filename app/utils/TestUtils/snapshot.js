/* eslint-disable import/no-extraneous-dependencies */
import renderer from 'react-test-renderer';

const mockSetState = Component =>
  class extends Component {
    setState(f, callback) {
      if (typeof f === 'function') {
        /* eslint-disable react/no-direct-mutation-state */
        this.state = { ...this.state, ...f(this.state) };
      } else {
        /* eslint-disable react/no-direct-mutation-state */
        this.state = { ...this.state, ...f };
      }
      if (callback) callback();
    }
  };

const assertSnapshot = (C, config) => {
  const { props, state, desc, throws, beforeTest } = config;

  let Component;
  let RenderedComponent;

  if (beforeTest) {
    beforeTest();
  }

  if (throws) {
    test(desc, () => {
      expect(() => {
        if (C.prototype.render) {
          Component = new C(props);
        } else {
          Component = C(props);
        }
      }).toThrow(throws);
    });
  } else {
    if (C.prototype.render) {
      const CWithMockState = mockSetState(C);
      Component = new CWithMockState(props);

      if (state) {
        Component.setState(state);
      }

      RenderedComponent = Component.render();
    } else {
      RenderedComponent = C(props);
    }

    test(desc, () => {
      const tree = RenderedComponent && renderer.create(RenderedComponent).toJSON();
      expect(tree).toMatchSnapshot();
    });
  }
};

const assertSnapshots = (Component, configs) => {
  configs.forEach(config => assertSnapshot(Component, config));
};

export { assertSnapshots, mockSetState };
export default { assertSnapshots, mockSetState };
