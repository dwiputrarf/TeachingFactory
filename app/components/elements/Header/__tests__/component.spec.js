import React from 'react';
import { View } from 'react-native';
import { assertSnapshots } from '../../../../utils/TestUtils/snapshot';
import Component from '../component';

jest.mock('../../BackButton', () => 'BackButton');
jest.mock('../../BurgerButton', () => 'BurgerButton');

describe('Component snapshot testing', () => {
  const configs = [
    {
      state: {},
      props: {
        burger: true,
        title: 'title'
      },
      desc: 'renders with burger'
    },
    {
      state: {},
      props: { back: true },
      desc: 'renders with back'
    },
    {
      state: {},
      props: { setting: true },
      desc: 'renders without data'
    },
    {
      state: {},
      props: {
        leftComponent: <View />,
        rightComponent: <View />,
        centerComponent: <View />
      },
      desc: 'renders without data'
    }
  ];
  assertSnapshots(Component, configs);
});
