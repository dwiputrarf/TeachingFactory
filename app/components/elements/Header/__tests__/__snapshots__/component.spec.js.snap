// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component snapshot testing renders with back 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#289F97",
      "flexDirection": "row",
      "height": 44,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
        },
        undefined,
      ]
    }
  >
    <BackButton />
  </View>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "flex": 4,
          "justifyContent": "center",
        },
        undefined,
      ]
    }
  />
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
        },
        undefined,
      ]
    }
  >
    <View />
  </View>
</View>
`;

exports[`Component snapshot testing renders with burger 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#289F97",
      "flexDirection": "row",
      "height": 44,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
        },
        undefined,
      ]
    }
  >
    <BurgerButton />
  </View>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "flex": 4,
          "justifyContent": "center",
        },
        undefined,
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#FFFFFF",
          "fontFamily": "Quicksand-Medium",
          "fontSize": 20,
        }
      }
    >
      title
    </Text>
  </View>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
        },
        undefined,
      ]
    }
  >
    <View />
  </View>
</View>
`;

exports[`Component snapshot testing renders without data 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#289F97",
      "flexDirection": "row",
      "height": 44,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
        },
        undefined,
      ]
    }
  >
    <View />
  </View>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "flex": 4,
          "justifyContent": "center",
        },
        undefined,
      ]
    }
  />
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
        },
        undefined,
      ]
    }
  >
    <RNSVGSvgView
      align="xMidYMid"
      bbHeight="48"
      bbWidth="48"
      height={48}
      meetOrSlice={0}
      minX={0}
      minY={0}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          undefined,
          false,
          Object {
            "flex": 0,
            "height": 48,
            "width": 48,
          },
        ]
      }
      vbHeight={48}
      vbWidth={48}
      width={48}
    >
      <RNSVGGroup
        fill={
          Array [
            0,
            0,
            0,
            0,
            1,
          ]
        }
        fillOpacity={1}
        fillRule={1}
        font={Object {}}
        matrix={
          Array [
            1,
            0,
            0,
            1,
            0,
            0,
          ]
        }
        opacity={1}
        originX={0}
        originY={0}
        propList={Array []}
        rotation={0}
        scaleX={1}
        scaleY={1}
        skewX={0}
        skewY={0}
        stroke={null}
        strokeDasharray={null}
        strokeDashoffset={null}
        strokeLinecap={0}
        strokeLinejoin={0}
        strokeMiterlimit={4}
        strokeOpacity={1}
        strokeWidth="1"
        x={0}
        y={0}
      >
        <RNSVGGroup
          fill={
            Array [
              0,
              1,
              1,
              1,
              1,
            ]
          }
          fillOpacity={1}
          fillRule={1}
          font={Object {}}
          matrix={
            Array [
              1,
              0,
              0,
              1,
              0,
              0,
            ]
          }
          opacity={1}
          originX={0}
          originY={0}
          propList={
            Array [
              "fill",
            ]
          }
          rotation={0}
          scaleX={1}
          scaleY={1}
          skewX={0}
          skewY={0}
          stroke={null}
          strokeDasharray={null}
          strokeDashoffset={null}
          strokeLinecap={0}
          strokeLinejoin={0}
          strokeMiterlimit={4}
          strokeOpacity={1}
          strokeWidth="1"
          x={0}
          y={0}
        >
          <RNSVGPath
            d="M28.85 30.05c-1.12.92-2.5 1.52-4.01 1.7m6.84-6.91c-.16 1.5-.75 2.88-1.64 4m1.64-5.69c-.16-1.5-.75-2.88-1.64-4m-1.19-1.2a7.754 7.754 0 0 0-4.01-1.71m-.91 1.42c-3.49 0-6.33 2.85-6.33 6.34s2.84 6.33 6.33 6.33 6.33-2.84 6.33-6.33-2.84-6.34-6.33-6.34zm-.78 14.1a7.715 7.715 0 0 1-4.06-1.65m-1.2-1.2a7.628 7.628 0 0 1-1.71-4.07m1.71-5.76a7.706 7.706 0 0 0-1.71 4.07m6.97-6.92c-1.52.15-2.92.75-4.06 1.66"
            fill={null}
            fillOpacity={1}
            fillRule={1}
            matrix={
              Array [
                1,
                0,
                0,
                1,
                0,
                0,
              ]
            }
            opacity={1}
            originX={0}
            originY={0}
            propList={
              Array [
                "stroke",
                "strokeWidth",
                "strokeLinecap",
                "strokeLinejoin",
                "strokeMiterlimit",
                "fill",
              ]
            }
            rotation={0}
            scaleX={1}
            scaleY={1}
            skewX={0}
            skewY={0}
            stroke={
              Array [
                0,
                1,
                1,
                1,
                1,
              ]
            }
            strokeDasharray={null}
            strokeDashoffset={null}
            strokeLinecap={1}
            strokeLinejoin={1}
            strokeMiterlimit={10}
            strokeOpacity={1}
            strokeWidth="0.7"
            x={0}
            y={0}
          />
          <RNSVGPath
            d="M23.15 16.23V14.8h1.69v1.44m0 15.51v1.45h-1.69v-1.44m5.7-13.81l1.05-1.06 1.2 1.2-1.06 1.06m-12.15 9.76l-.99.99 1.19 1.2 1-.99m12.59-6.96h1.52v1.69h-1.52m-15.5 0H14.8v-1.69h1.38m13.86 5.69l1.06 1.06-1.2 1.2-1.05-1.05m-9.76-12.16l-1-1-1.19 1.2.99.99"
            fill={
              Array [
                0,
                1,
                1,
                1,
                1,
              ]
            }
            fillOpacity={1}
            fillRule={1}
            matrix={
              Array [
                1,
                0,
                0,
                1,
                0,
                0,
              ]
            }
            opacity={1}
            originX={0}
            originY={0}
            propList={
              Array [
                "stroke",
                "strokeWidth",
                "strokeLinecap",
                "strokeLinejoin",
                "strokeMiterlimit",
                "fill",
              ]
            }
            rotation={0}
            scaleX={1}
            scaleY={1}
            skewX={0}
            skewY={0}
            stroke={
              Array [
                0,
                1,
                1,
                1,
                1,
              ]
            }
            strokeDasharray={null}
            strokeDashoffset={null}
            strokeLinecap={1}
            strokeLinejoin={1}
            strokeMiterlimit={10}
            strokeOpacity={1}
            strokeWidth="0.7"
            x={0}
            y={0}
          />
        </RNSVGGroup>
      </RNSVGGroup>
    </RNSVGSvgView>
  </View>
</View>
`;

exports[`Component snapshot testing renders without data 2`] = `
<View
  style={
    Object {
      "backgroundColor": "#289F97",
      "flexDirection": "row",
      "height": 44,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
        },
        undefined,
      ]
    }
  >
    <View />
  </View>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "flex": 4,
          "justifyContent": "center",
        },
        undefined,
      ]
    }
  >
    <View />
  </View>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
        },
        undefined,
      ]
    }
  >
    <View />
  </View>
</View>
`;
